community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java

```diff
@@ -92,7 +91,11 @@ public boolean next()
 
         if ( !payload.next() )
         {
-            throw new NotFoundException( "Property record with id " + currentPropertyRecordId + " not in use" );
+            if ( nextPropertyRecordId != Record.NO_NEXT_PROPERTY.intValue() )
+            {
+                return next();
```

Comment:
This looks like a recursive call, one that could potentially be very deep. The case would be reading the first record in a property chain | another tx deleting the whole chain (a really really long chain of, say, short strings where each string occupies a whole property record, 4 blocks) | trying to read the next property would recurse <number of properties on this entity> times. We don't have to have recursion here, do we?